server/routers/deliverables.js

```diff
@@ -19,55 +35,62 @@ router.post('/add', (req, res) => {
     else if (description.length > 200) {
         res.status(400).send(apiError.InvalidRequest);
     }
-    else if(link ===null){
+    else if(url ===null){
         res.status(400).send(apiError.InvalidRequest);
     }
-    //creating the object
-    let deliverable = {
-        projectId,
-        title,
-        description,
-        link
+    else {
+        const checkProject = await Project.findOne({ where: { id: project_id } })
+        if (checkProject === null) {
+            res.status(400).send({ "message": "Project not found" })
+        }
     }
 
-    //storing it into the database
-
-
-    //response (will be a message after the db is made)
-    res.status(200).send(deliverable)
-})
-
-router.get('/get', (req, res) => {
-    projectId = req.body.projectId
+    try {
+        const del = await Deliverable.create({ title, description, url, project_id })
 
-    //search in db
+        res.status(200).send({ "message": del.title + " was created." })
+    }
+    catch (err) {
+        console.log(err)
+        res.status(400).send({ "message": "Something bad happened" })
+    }
 
+    try {
+        const del = await Deliverable.create({ title, description, url, project_id })
 
-    res.status(200).send(projectId)
+        res.status(200).send({ "message": del.title + " was created." })
+    }
+    catch (err) {
+        console.log(err)
+        res.status(400).send({ "message": "Something bad happened" })
+    }
 })
 
-router.put('/update', (req, res) => {
-    projectId = req.body.projectId
+router.put('/:id', async (req, res) => {
+    id = req.params.id
     title = req.body.title
     description = req.body.description
     link = req.body.link
-    //validate projectId
-
-
-    if (title !== null && title.length < 10) {
-        //update title in the db
+    project_id = req.body.project_id
 
+    let checkDeliverable = await Deliverable.findOne({ where: { id, project_id } })
```

Comment:
honestly can't believe this works and you don't need to specify `{ id: id, project_id: project_id }`

Reply:
If the variables have the variable has the same name with the property, it works. It's way easier to write like this.